sqlmesh/core/context.py

```diff
@@ -479,6 +479,17 @@ def scheduler(self, environment: t.Optional[str] = None) -> Scheduler:
         if not snapshots:
             raise ConfigError("No models were found")
 
+        return self._create_scheduler(snapshots)
+
+    def _create_scheduler(self, snapshots: t.Iterable[Snapshot]) -> Scheduler:
```

Comment:
Should we just make it public (no `_`)?

Reply:
Yeah, we can. My only concern was having it and `def scheduler` method both be public, it could be confusing. I don't have a strong opinion though. Do you?